plugin/procfs/procfs.cc

```diff
@@ -0,0 +1,475 @@
+                        in_func->arguments()[0]->full_name(),
+                        field_name.c_str()) == 0)
+      {
+        char    buff[1024];
+        String *res;
+        String  filter(buff, sizeof(buff), system_charset_info);
+        for (uint i= 1; i < in_func->arg_count; ++i)
+        {
+          if (in_func->arguments()[i] != NULL &&
+              (res= in_func->arguments()[i]->val_str(&filter)) &&
+              res->length() > 0)
+          {
+            in_args[std::string(res->c_ptr_safe(), res->length())]= true;
+          }
+        }
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+static void fill_procfs_view_row(THD *thd, TABLE *table, const char *fname,
+                                 char *buf, size_t sz)
+{
+  if (sz == 0)
+    return;
+
+  table->field[0]->store(fname, strlen(fname), system_charset_info);
+  table->field[1]->store(buf, sz, system_charset_info);
+  schema_table_store_record(thd, table);
+}
+
+static std::vector<std::string> limited_glob_files(std::string path,
+                                                   std::string pattern,
+                                                   int         max_results)
+{
+  std::vector<std::string> files_found;
+
+  if (max_results <= 0)
+    return files_found;
+
+  DIR *          dir= opendir(path.c_str());
+  struct dirent *dir_entry;
+
+  if (!dir)
+    return files_found;
+
+  std::vector<char> real_file_path(PATH_MAX);
+
+  while ((dir_entry= readdir(dir)) != 0)
+  {
+    if (!fnmatch(pattern.c_str(), dir_entry->d_name,
+                 FNM_FILE_NAME | FNM_CASEFOLD | FNM_NOESCAPE | FNM_PERIOD))
+    {
+      if (dir_entry->d_type != DT_DIR)
+      {
+        std::string fname=
+            std::string(path).append("/").append(dir_entry->d_name);
+
+        if (realpath(fname.c_str(), real_file_path.data()))
+        {
+          if (strncmp("/sys/", real_file_path.data(), strlen("/sys/")) == 0 ||
+              strncmp("/proc/", real_file_path.data(), strlen("/proc/")) == 0)
+            files_found.push_back(fname);
+        }
+        if (files_found.size() >= static_cast<size_t>(max_results))
+          break;
+      }
+    }
+  }
+
+  closedir(dir);
+  return files_found;
+}
+
+static std::vector<std::string> limited_glob(std::string query,
+                                             int max_results, int max_depth)
+{
+  std::vector<std::string> files_found;
+
+  if (max_results <= 0 || max_depth <= 0)
+    return files_found;
+
+  std::string path;
+  std::string pattern;
+  std::string rest;
+
+  std::string::size_type first_star= query.find_first_of("*?[");
+  std::string::size_type last_sep_before_star=
+      query.find_last_of("/", first_star);
+  std::string::size_type first_sep_after_star=
+      query.find_first_of("/", first_star);
+
+  if (last_sep_before_star == std::string::npos)
+  {
+    path= std::string(".");
+    pattern= query;
+  }
+  else if (first_sep_after_star == std::string::npos)
```

Comment:
How is this case possible? If <is not npos, then we have a separator, which means first_sep_after_star will be smaller or equal to last_sep_before_star, it can't be npos.